include/dlaf/mc/triangular_solve_distributed.h

```diff
@@ -0,0 +1,268 @@
+                                                                tile);
+                      return std::move(tile);
+                    }),
+                    k_rank_row, mat_b.tileSize(GlobalTileIndex(k, j)), serial_comm());
+              }
+            }
+
+          }  // j_local loop
+
+          for (SizeType i_local = distr_a.nextLocalTileFromGlobalTile<Coord::Row>(k + 1);
+               i_local < localnrtile_rows; ++i_local) {
+            auto i = distr_a.globalTileFromLocalTile<Coord::Row>(i_local);
+
+            // Choose queue priority
+            auto trailing_executor = (i == k + 1) ? executor_hp : executor_normal;
+
+            hpx::shared_future<Tile<const T, Device::CPU>> ik_tile;
+
+            // Broadcast Aik row-wise
+            if (mat_a.rankIndex().col() == k_rank_col) {
+              auto k_local_col = distr_a.localTileFromGlobalTile<Coord::Col>(k);
+
+              auto ik = LocalTileIndex{i_local, k_local_col};
+
+              if (row_comm_size > 1) {
+                hpx::dataflow(hpx::util::unwrapping([](auto&& tile, auto&& comm_wrapper) {
+                                dlaf::comm::sync::broadcast::send(comm_wrapper().rowCommunicator(),
+                                                                  tile);
+                              }),
+                              mat_a.read(ik), serial_comm());
+              }
+              ik_tile = mat_a.read(ik);
+            }
+            else {
+              if (row_comm_size > 1) {
+                ik_tile =
+                    hpx::dataflow(hpx::util::unwrapping(
+                                      [](auto index, auto&& tile_size,
+                                         auto&& comm_wrapper) -> Tile<const T, Device::CPU> {
+                                        memory::MemoryView<T, Device::CPU> mem_view(
+                                            util::size_t::mul(tile_size.rows(), tile_size.cols()));
+                                        Tile<T, Device::CPU> tile(tile_size, std::move(mem_view),
+                                                                  tile_size.rows());
+                                        dlaf::comm::sync::broadcast::receive_from(index,
+                                                                                  comm_wrapper()
+                                                                                      .rowCommunicator(),
+                                                                                  tile);
+                                        return std::move(tile);
+                                      }),
+                                  k_rank_col, mat_a.tileSize(GlobalTileIndex(i, k)), serial_comm());
+              }
+            }
+
+            for (SizeType j_local = 0; j_local < localnrtile_cols; ++j_local) {
+              // Matrix multiplication to update other eigenvectors
```

Comment:
Eigenvector back-transformation is a specific application of the triangular solver.
In general there is no eigenvectors involved.